### PR TITLE
controller: StorageClass change should trigger the controller sync-up

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn/longhorn#3821

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

File: controller/kubernetes_configmap_controller.go [hunk 1]
```diff
@@ -234,3 +245,26 @@ func (kc *KubernetesConfigMapController) enqueueConfigMapChange(obj interface{})
...
+		}
+
+		sc, ok = deletedState.Obj.(*storagev1.StorageClass)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("DeletedFinalStateUnknown contained non StorageClass object: %#v", deletedState.Obj))
+			return false
+		}
+	}
+
+	return sc.Provisioner == types.LonghornDriverName
```
Comment (PhanLe1010) [inline controller/kubernetes_configmap_controller.go:269]: Can we filter out other`driver.longhorn.io` SC and only consider the `longhorn` SC with `driver.longhorn.io`. e.g:
```
return sc.Provisioner == types.LonghornDriverName && sc.Name == type.DefaultStorageClassName
```

Comment (PhanLe1010) [inline controller/kubernetes_configmap_controller.go:269]: Can we filter out other`driver.longhorn.io` SC and only consider the `longhorn` SC with `driver.longhorn.io`. e.g:
```
return sc.Provisioner == types.LonghornDriverName && sc.Name == type.DefaultStorageClassName
```

Comment (PhanLe1010) [reply controller/kubernetes_configmap_controller.go:269]: Ops. This is already merged

Comment (shuo-wu) [reply controller/kubernetes_configmap_controller.go:269]: No worries. Let me create one more PR for this.

Comment (PhanLe1010) [reply controller/kubernetes_configmap_controller.go:269]: Thank you

File: controller/kubernetes_configmap_controller.go [hunk 2]
```diff
@@ -73,6 +73,17 @@ func NewKubernetesConfigMapController(
...
+		cache.FilteringResourceEventHandler{
+			FilterFunc: isLonghornStorageClass,
+			Handler: cache.ResourceEventHandlerFuncs{
+				UpdateFunc: func(old, cur interface{}) { kc.enqueueConfigMapForStorageClassChange(cur) },
+				DeleteFunc: kc.enqueueConfigMapForStorageClassChange,
+			},
+		},
+	)
+
 	kc.cacheSyncs = append(kc.cacheSyncs, ds.ConfigMapInformer.HasSynced)
```
Comment (joshimoo) [inline controller/kubernetes_configmap_controller.go:87]: @shuo-wu We are missing the wait for cache sync for the storage class informer.

Comment (shuo-wu) [reply controller/kubernetes_configmap_controller.go:87]: Sorry for missing it! 🤣

**Pull request merged**